### ISSUE TITLE
link ai server in Ricerca Globale

### ISSUE THREAD

**Issue by enricodarodda** [Stream4me/addon#191]
Buonasera,
premetto che non so se sia un problema solamente mio o di qualche impostazione sbagliata, ma i risultati che vengono presentati nella ricerca globale, sia film che serie tv sono sbagliati, ovvero ci sono i link che riguardano la ricerca effettuata ma ce ne sono anche altri completamente non correlati e non comprendo a cosa possa essere dovuto tutto ciò.
Grazie in anticipo e resto a disposizione per ulteriori chiarimenti.

**Comment by Alhaziel01** [comment]
Sono I risultati che restituiscono i vari siti, se provi a cercare su uno di quelli che ti presentano risultati sballati, vedrai che combaciano.

**Comment by enricodarodda** [comment]
ah ok, avendo quindi io impostato la vista dei canali in "tutto insieme", vedo anche i risultati sbagliati restituiti, giusto? Sarebbe possibile fare un filtro a posteriori, quindi dopo aver ottenuto i risultati, filtrare nuovamente per il valore della ricerca iniziale? In modo da evitare che risultati sbagliati siano presenti nell'elenco?

**Comment by enricodarodda** [comment]
Poi volevo chiedere, dato che non so se sia una scelta o una cosa programmata per il futuro magari, ma c'è la possibilità di vedere direttamente una lista dei link finali al posto dei canali con i risultati all'interno o al posto proprio dei resultati? (intendo proprio il link a openload o wstream della situazione)
Grazie in anticipo di nuovo

**Comment by mac12m99** [comment]
Ciao, questa funzionalità è stata appena aggiunta nella versione beta ed è quindi da considerarsi in fase di test, se vuoi provarla.
Per attivarla vai nelle impostazioni della ricerca e seleziona "mostra risultati" -> "tutti insieme", poi fai "cerca film" (con le altre modalità non funziona e non potrà mai funzionare)

**Comment by enricodarodda** [comment]
La ringrazio inanzitutto per aver preso in considerazione quanto proposto in questo thread e per avermi avvisato dell'enhancement. Proverò di sicuro la nuova feature.
Mi scuso se mi permetto di chiederle, ma quando dice che con le altre modalità non potrà mai funzionare significa che l'implementazione modificherebbe completamente la struttura dell'addon e non è gestibile o c'è qualche altro motivo che ignoro?
Grazie mille

**Comment by mac12m99** [comment]
Di nulla,
con quella frase mi riferivo al fatto che sarebbe impossibile fare lo stesso anche con "cerca per titolo" (perchè non si sa a priori quale contenuto si intende guardare) e "cerca serie" (perchè, essendo composte da episodi non avrebbe senso, anche perchè mostrare tutti i server per ogni episodio per ogni canale renderebbe la ricerca oltremodo lenta)

**Comment by enricodarodda** [comment]
Io facevo riferimento alla gestione dei link finali come li gestisce un altro addon, come ad esempio exodus, che è vero non ha la ricerca globale, ma per esempio anche nelle serie tv riesce a ricondurre i link alle singole puntate, quindi presumo che abbai una struttura già impostata delle serie tv e delle stagioni/puntate e poi vada a recuperare i vari link per le puntate. Non so se sia possibile o abbia senso (questo è solo il mio modo di vedere le cose): analizzare tutti i vari link che vengono proposti dai vari canali (seriehd, ecc...)(come già per altro viene fatto se non sbaglio) avere una base di stagioni/puntate recuperate da un qualche database (non so se esista o se sia fattibile, ma credo che exodus faccia questo) e poi in base a tutti i risultati ottenuti per ogni canale recuperare solamente i link della puntata selezionata, quindi si tratterebbe solamente di un filtro rispetto alle richieste già effettuate ai vari canali. Se invece non fosse gestibile una cosa del genere di sicuro è meglio la struttura attuale poichè più ordinata anzichè avere tutte le stagioni e puntate messe "a caso" in quanto sequenza di link e basta.
(con questo non voglio assolutamente far confronti tra addon e dire se uno è meglio dell'altro o cose del genere, volevo solo riportare la mia esperienza)
Grazie mille per il suo tempo

**Comment by mac12m99** [comment]
Gestire le cose come le hai descritte è fattibile, anche se servirebbe un lavoro extra per supportare l'aggiunta in videoteca e il download.
Più che altro per quanto riguarda i film è utile perchè ti fa risparmiare un click e ti da una visione d'insieme, ma per le serie io personalmente uso la ricerca solo per aggiungere in videoteca, in modo avere le puntate già salvate e direttamente nel menu di kodi.
Poi c'è gente che non sa della videoteca e continuerà ad oltranza a cercare la stessa serie ogni volta che vuole vedere un episodio, e per questi una struttura del genere sarebbe comoda..

ad un utente "che sa quello che fa" può essere utile solo per sapere al volo quali canali hanno tutte le stagioni e eventualmente riprodurre la prima puntata.

In sostanza fattibile si, utile non tantissimo ma comunque visto che ormai ci sono vedrò di sperimentare qualcosa, se non ci perdo troppo tempo..

**Comment by enricodarodda** [comment]
Concordo con lei, la videoteca è sicuramente una funzione utilissima e anche io utilizo la ricerca solamente la prima volta, ma la cosa che mi sono ritrovato a fare (ed è il motivo principale per cui le richiedevo la funzionalità), è perdere un pò di tempo all'inizio a cercare i link giusti diciamo, perchè non sempre i link funzionano o non tutti i canali che si definiscono hd sono effettivamente hd, perciò si perde più tempo inizialmente a cercare il canale giusto. In aggiunta può capitare che  i link delle serie salvate in videoteca cambino e quindi tocchi fare nuovamente la ricerca.
Ad ogni modo la ringrazio di nuovo per avermi ascoltato e preso in considerazione la proposta. Grazie anche per trovare del tempo per sperimentare, ma come ha detto lei, solamente se non le porta via troppo tempo, poichè mi rendo conto che la quantità di tempo richiesta non sarebbe indifferente.
Cordiali Saluti
